### PR TITLE
Add resource hints for Condor provider

### PR DESCRIPTION
**Pull request by annawoodard**
Part of fix for #942.

### CODE REVIEW

File: parsl/providers/condor/condor.py [hunk 1]
```diff
@@ -187,6 +198,16 @@ def submit(self, command, tasks_per_node, job_name="parsl.auto"):
...
         job_name = "parsl.{0}.{1}".format(job_name, time.time())
 
+        scheduler_options = self.scheduler_options
+        worker_init = self.worker_init
+        if self.mem_per_node is not None:
+            scheduler_options += 'RequestMemory = {}'.format(self.mem_per_node * 1024)
+            worker_init += 'export PARSL_MEMORY_GB={}\n'.format(self.mem_per_node)
+        if self.cores_per_node is not None:
+            cpus_per_task = math.floor(self.cores_per_node / tasks_per_node)
+            scheduler_options += 'RequestCpus = {}'.format(cpus_per_task)
```
[the remark this answers is not in the document]
Comment (benclifford) [reply parsl/providers/condor/condor.py]: > we cannot expect the user to know the cores_per_node

This comment highlights a documentation problem that I have had problems with in the HPC / MPI world for decades: are these parameters *describing* what the remote system looks like (i.e. describing the hardware purchasing decision made by some committee), or are they controlling what parsl will *request* from the remote system.

[the remark this answers is not in the document]
Comment (annawoodard) [reply parsl/providers/condor/condor.py]: > or are they controlling what parsl will request from the remote system.
In this particular case, it is the latter. I will clarify in the docstring.

File: parsl/providers/condor/condor.py [hunk 2]
```diff
@@ -187,6 +197,15 @@ def submit(self, command, tasks_per_node, job_name="parsl.auto"):
 
         job_name = "parsl.{0}.{1}".format(job_name, time.time())
 
+        scheduler_options = self.scheduler_options
+        worker_init = self.worker_init
+        if self.mem_per_node is not None:
+            scheduler_options += 'RequestMemory = {}'.format(self.mem_per_node * 1024)
```
Comment (benclifford) [inline parsl/providers/condor/condor.py]: Should there be `\n` on the end of this and line 206?

Comment (annawoodard) [reply parsl/providers/condor/condor.py]: Good catch-- I had changed that locally while testing and forgotten to push.

Comment (annawoodard) [reply parsl/providers/condor/condor.py]: It's fixed now

**Pull request merged**